src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -30,18 +32,28 @@ internal class CSharpAsAndNullCheckDiagnosticAnalyzer : AbstractCodeStyleDiagnos
 
         public CSharpAsAndNullCheckDiagnosticAnalyzer()
             : base(IDEDiagnosticIds.InlineAsTypeCheckId,
-                   new LocalizableResourceString(
-                       nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+                new LocalizableResourceString(
+                    nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
         {
         }
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
+            => context.RegisterSyntaxNodeAction(SyntaxNodeAction,
+                SyntaxKind.IfStatement, SyntaxKind.WhileStatement);
 
         private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
         {
+            var node = syntaxContext.Node;
+            var syntaxTree = node.SyntaxTree;
+
+            // "x is Type y" is only available in C# 7.0 and above. Don't offer this refactoring
+            // in projects targeting a lesser version.
+            if (((CSharpParseOptions)syntaxTree.Options).LanguageVersion < LanguageVersion.CSharp7)
```

Comment:
Tagging @mavasani .  It seems unfortunate that we will get tons of callbacks for a file and we'll have to do this check and bail out for every node we get called back on.  It would be really nice if we could have some sort of conditional way to say "don't even call into us for this syntax tree."